### ISSUE TITLE
Browserify example doesn't find bitcoinjs

### ISSUE THREAD

**Issue by tgflynn** [bitcoinjs/bitcoinjs-lib#593]
The instructions on the web page for creating a js bundle with browserify don't work on Ubuntu 14.04 LTS.

Specifically after running the commands:

sudo npm install -g bitcoinjs-lib
sudo npm -g install bitcoinjs-lib browserify uglify-js

(both of these appear to complete normally)

browserify -r bitcoinjs-lib -s bitcoin | uglifyjs > bitcoinjs.min.js

fails with the following error message:

Error: Cannot find module 'bitcoinjs-lib' from '/home/flynn/projects/bitcoin/sigtool'
    at /usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:46:17
    at process (/usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:173:43)
    at ondir (/usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:188:17)
    at load (/usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:69:43)
    at onex (/usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:92:31)
    at /usr/local/lib/node_modules/browserify/node_modules/resolve/lib/async.js:22:47
    at FSReqWrap.oncomplete (fs.js:82:15)

node --version
v4.4.5

npm --version
2.15.5

browserify --version
13.0.1

I strace'd the browserify command and it doesn't even attempt to open any files containing bitcoin in the path.

I also tried the foobar.js example from the README and this also does not appear to work because the generated foobar.js contains only the following:

(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.foobar = f()}})(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){

},{}]},{},[1])(1)
});

I guess this might be a bug in browserify but I'm initially reporting it here because it's the bitcoinjs installation instructions that fail.

**Comment by dcousens** [comment]
@tgflynn I'll investigate this tomorrow,  but I can assure you `bitcoinjs-lib` does still work with browserify,  it may just be something wrong with this specific example.

**Comment by dcousens** [comment]
``` js
browserify -r bitcoinjs-lib -s bitcoin
```

Worked for me,   it may not be finding the globally installed `bitcoinjs-lib`... does `browserify` even look at global dependencies?  I used locally installed dependencies (aka, an adjacent `node_modules/bitcoinjs-lib` existed).

Otherwise, are you sure `bitcoinjs-lib` installed correctly?

**Comment by tgflynn** [comment]
> ... does browserify even look at global dependencies?

I don't know I hadn't used it before.  I just followed the instructions under Get Started/Compile for Browser on the bitcoinjs web site and found that they didn't work.  Those instructions don't say anything about creating a local repository.  They suggest that you can just use browserify to create a bundle for bitcoinjs that can be used from the browser.

After not getting bitcoinjs to work I tried cryptocoinjs and found that it was indeed necessary to create a local repository with npm init then install dependencies locally into that repository with npm install before browserify could be used successfully.  So I suspect that it's true that browserify only works with a local repository and that the bitcoinjs instructions under Get Started... are misleading.

**Comment by dcousens** [comment]
@tgflynn thanks,  I've [made an issue](https://github.com/bitcoinjs/bitcoinjs.github.com/issues/7) to ensure that gets updated.

**Comment by dcousens** [comment]
@tgflynn would you be able to elaborate on what exactly you did?  That we can document that for others?

**Comment by tgflynn** [comment]
You mean what I did with bitcoinjs that failed ?

I just followed the instructions on the website **literally** after installing nodejs and npm - first the default Ubuntu 14.04 version then the node LTS version v4.4.5 thinking it might be a version problem.

**Comment by dcousens** [comment]
@tgflynn I meant,  what did you end up doing that worked for you.    So if others come along and find this issue,  it might help them.
Also,  based on how you frame what worked for you,   I might edit the site to be more helpful along similar lines.

**Comment by tgflynn** [comment]
Well at this point I'm using cryptocoinjs so what I'm doing may not be terribly relevant for bitcoinjs-lib.

Also I haven't actually got my app working yet (I had abandoned it because I've been underbid anyway but I've decided to complete the task if only for the learning experience).  At the moment I'm still trying to figure out exactly how browserify is supposed to be used.  I do have some kind of bundle but it's not immediately obvious how to actually call it from a web page.

My suggestion would be to add a link to a good browserify tutorial and either not provide specific commands for using browserify or else provide a complete example of how to do that including all necessary steps.

At the moment the closest thing I've seen to such an example is this: 

http://cryptocoinjs.com/guide/getting-started/

and that's basically what I did to get a cryptocoinjs bundle.  Even this example is a bit lacking though because it doesn't show you how to use the resulting bundle from a web page, only how to use it under node, which I don't think you would need browserify for.  I also had to guess my way through the various questions asked by npm init since I don't yet understand how or if those affect browserify.
